pkgs/build-support/fetchurl/default.nix

```diff
@@ -1,4 +1,4 @@
-{ stdenv, curl }: # Note that `curl' may be `null', in case of the native stdenv.
+{ stdenv, curl, ca_cert_file ? null, insecure ? false }: # Note that `curl' may be `null', in case of the native stdenv.
```

Comment:
The performance penalty should be minute compared to the costs of
everything else in the TLS stack.

On Sun, May 31, 2015, 00:54 William Kennington <william@wkennington.com>
wrote:

> Well, the main reason I want this implemented is for updated packages so
> you can reasonably trust hashes you get out of failed fetchurls.
>
> On Sun, May 31, 2015, 00:53 Vladimír Čunát <notifications@github.com>
> wrote:
>
>> In pkgs/build-support/fetchurl/default.nix
>> <https://github.com/NixOS/nixpkgs/pull/8082#discussion_r31387880>:
>>
>> > @@ -1,4 +1,4 @@
>> > -{ stdenv, curl }: # Note that `curl' may be `null', in case of the native stdenv.
>> > +{ stdenv, curl, ca_cert_file ? null, insecure ? false }: # Note that `curl' may be `null', in case of the native stdenv.
>>
>> Was there a discussion about (dis)advantages of using insecure connection
>> by default for fixed-output derivations? For most use cases I don't think
>> there's any added security, and I suppose there's some (small) performance
>> loss.
>>
>> —
>> Reply to this email directly or view it on GitHub
>> <https://github.com/NixOS/nixpkgs/pull/8082/files#r31387880>.
>>
>